NuGetProtocolSamples/Samples.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using NuGet.Common;
+using NuGet.Packaging;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using Xunit;
+
+namespace NuGet.Protocol.Samples
+{
+    public class Samples
+    {
+        [Fact]
+        public async Task ListPackageVersions()
```

Comment:
Most samples I've seen are console apps, just curious why we picked tests here.

Reply:
Several reasons:

1. It's easy to bundle many samples in one file
1. You can do a quick (though not perfect) sanity check for broken samples with `dotnet test`
1. It follows the Azure SDK samples pattern

Reply:
When you run `dotnet test` is the console output visible?

Reply:
Talked offline. We decided to go with a console app so that users can `dotnet run` this and see meaningful output.